// Copyright 2025 The Ray Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//  http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option cc_enable_arenas = true;
package ray.rpc;

// This message defines the event_data stored by the export API for
// EXPORT_DATASET_OPERATOR_SCHEMA type events from Ray Data operators.
message ExportDatasetOperatorSchema {
  // The operator UUID
  string operator_uuid = 1;

  // Schema as a dictionary mapping field names to their types (both strings)
  map<string, string> schema_fields = 2;
}
